crates/meilisearch/tests/auth/authorization.rs

```diff
@@ -10,73 +10,103 @@ use time::{Duration, OffsetDateTime};
-            ("PUT",     "/indexes/products/settings/sortable-attributes") =>   hashset!{"settings.update", "settings.*", "*"},
-            ("PUT",     "/indexes/products/settings/stop-words") =>            hashset!{"settings.update", "settings.*", "*"},
-            ("PUT",     "/indexes/products/settings/synonyms") =>              hashset!{"settings.update", "settings.*", "*"},
-            ("GET",     "/indexes/products/stats") =>                          hashset!{"stats.get", "stats.*", "*"},
-            ("GET",     "/stats") =>                                           hashset!{"stats.get", "stats.*", "*"},
-            ("POST",    "/dumps") =>                                           hashset!{"dumps.create", "dumps.*", "*"},
-            ("POST",    "/snapshots") =>                                       hashset!{"snapshots.create", "snapshots.*", "*"},
-            ("GET",     "/version") =>                                         hashset!{"version", "*"},
-            ("GET",     "/metrics") =>                                         hashset!{"metrics.get", "metrics.*", "*"},
-            ("POST",    "/logs/stream") =>                                     hashset!{"metrics.get", "metrics.*", "*"},
-            ("DELETE",  "/logs/stream") =>                                     hashset!{"metrics.get", "metrics.*", "*"},
-            ("PATCH",   "/keys/mykey/") =>                                     hashset!{"keys.update", "*"},
-            ("GET",     "/keys/mykey/") =>                                     hashset!{"keys.get", "*"},
-            ("DELETE",  "/keys/mykey/") =>                                     hashset!{"keys.delete", "*"},
-            ("POST",    "/keys") =>                                            hashset!{"keys.create", "*"},
-            ("GET",     "/keys") =>                                            hashset!{"keys.get", "*"},
-            ("GET",     "/experimental-features") =>                           hashset!{"experimental.get", "*"},
-            ("PATCH",   "/experimental-features") =>                           hashset!{"experimental.update", "*"},
-            ("GET",   "/network") =>                                           hashset!{"network.get", "*"},
-            ("PATCH",   "/network") =>                                         hashset!{"network.update", "*"},
-        };
-
-        authorizations
-    });
+macro_rules! hashset {
+    ( $( $val:tt ),* $(,)? ) => {{
+        let mut set: HashSet<&'static [&'static str]> = HashSet::new();
+        $(
+            hashset!(@insert set, $val);
+        )*
+        set
+    }};
+
+    // Match array-like input: ["a", "b"]
+    (@insert $set:ident, [ $($elem:literal),* ]) => {{
+        const ITEM: &[&str] = &[$($elem),*];
+        $set.insert(ITEM);
+    }};
+
+    // Match single literal: "a"
+    (@insert $set:ident, $val:literal) => {{
+        const ITEM: &[&str] = &[$val];
+        $set.insert(ITEM);
+    }};
+}
```

Comment:
Previously we used a macro producing `HashSet<&str>` from a list of actions. The logic now has to use `HashSet<&[&str]>` but I didn't want to rewrite everything and wrap everything in Vecs, so I modified the macro so that it accepts both arrays and single values, wrapping them in a new array automatically